lib/jxl/decode_test.cc

```diff
@@ -2986,135 +2986,144 @@ TEST(DecodeTest, SkipFrameWithAlphaBlendingTest) {
+        EXPECT_EQ(JXL_DEC_SUCCESS,
+                  JxlDecoderGetFrameHeader(dec, &frame_header));
+        size_t buffer_size;
+        EXPECT_EQ(JXL_DEC_SUCCESS,
+                  JxlDecoderImageOutBufferSize(dec, &format, &buffer_size));
+        if (coalescing)
+          EXPECT_EQ(xsize * ysize * 8, buffer_size);
+        else
+          EXPECT_EQ(
+              frame_header.layer_info.xsize * frame_header.layer_info.ysize * 8,
+              buffer_size);
+        frames[i].resize(buffer_size);
+        EXPECT_EQ(JXL_DEC_SUCCESS,
+                  JxlDecoderSetImageOutBuffer(dec, &format, frames[i].data(),
+                                              frames[i].size()));
+        EXPECT_EQ(JXL_DEC_FULL_IMAGE, JxlDecoderProcessInput(dec));
+        EXPECT_EQ(frame_header.layer_info.blend_info.blendmode,
+                  JXL_BLEND_REPLACE);
+        if (coalescing) {
+          EXPECT_EQ(frame_header.layer_info.xsize, oxsize);
+          EXPECT_EQ(frame_header.layer_info.ysize, oysize);
+          EXPECT_EQ(frame_header.layer_info.crop_x0, 0);
+          EXPECT_EQ(frame_header.layer_info.crop_y0, 0);
+        } else {
+          // manually merge this layer
+          int x0 = frame_header.layer_info.crop_x0;
+          int y0 = frame_header.layer_info.crop_y0;
+          int w = frame_header.layer_info.xsize;
+          int h = frame_header.layer_info.ysize;
+          for (int y = 0; y < static_cast<int>(oysize); y++) {
+            if (y < y0 || y >= y0 + h) continue;
+            // pointers do whole 16-bit RGBA pixels at a time
+            uint64_t* row_merged = static_cast<uint64_t*>(
+                (void*)(frames[4].data() + y * oxsize * 8));
+            uint64_t* row_layer = static_cast<uint64_t*>(
+                (void*)(frames[i].data() + (y - y0) * w * 8));
+            for (int x = 0; x < static_cast<int>(oxsize); x++) {
+              if (x < x0 || x >= x0 + w) continue;
+              row_merged[x] = row_layer[x - x0];
             }
           }
-
-          // After all frames were decoded, JxlDecoderProcessInput should return
-          // success to indicate all is done.
-          EXPECT_EQ(JXL_DEC_SUCCESS, JxlDecoderProcessInput(dec));
-          JxlThreadParallelRunnerDestroy(runner);
-          JxlDecoderDestroy(dec);
         }
       }
 
-      EXPECT_EQ(0u, jxl::test::ComparePixels(frames[0].data(), frames[4].data(),
-                                             oxsize, oysize, format, format));
+      // After all frames were decoded, JxlDecoderProcessInput should return
+      // success to indicate all is done.
+      EXPECT_EQ(JXL_DEC_SUCCESS, JxlDecoderProcessInput(dec));
+      JxlThreadParallelRunnerDestroy(runner);
+      JxlDecoderDestroy(dec);
+    }
+
+    EXPECT_EQ(0u, jxl::test::ComparePixels(frames[0].data(), frames[4].data(),
+                                           oxsize, oysize, format, format));
+  };
+
+  for (bool keep_orientation : {true, false}) {
```

Comment:
Not necessary for now, but could we start converting some of those nested for loops into parameterized tests?

Reply:
Ack. Sure, make sense.